palamar/views.py

```diff
@@ -1,8 +1,16 @@
 import datetime
 
 from django.contrib.auth.decorators import login_required
-from django.shortcuts import render
+from django.shortcuts import render, redirect
+
+from user_profile.models import Profile
 
 @login_required
 def index(request):
+    user_profile = Profile.objects.filter(user=request.user)
+
+    if user_profile[0].selected_project < 1:
```

Comment:
Checking if user has a project should not be made with user_profile.selected_project model. You should check if use has any role assignment to any project.

Because if user does not have any selected_project but the current user may have some role in some project. So it will be better to check if user has any role assignment to any project.

Please check Role and Assignment models.